src/builtin.c

```diff
@@ -548,6 +564,38 @@ static jv f_format(jq_state *jq, jv input, jv fmt) {
     }
     jv_free(input);
     return line;
+  } else if (!strcmp(fmt_s, "base64d")) {
+    jv_free(fmt);
+    input = f_tostring(jq, input);
+    const unsigned char* data = (const unsigned char*)jv_string_value(input);
+    int len = jv_string_length_bytes(jv_copy(input));
+    char result[(3 * len) / 4]; // decoded result buffer
+    int ri = 0;
+    int input_bytes_read=0;
+    uint32_t code = 0;
+    for (int i=0; i<len && data[i] != '='; i++) {
+      code <<= 6;
+      code |= BASE64_DECODE_TABLE[data[i]];
```

Comment:
If the input isn't base64, shouldn't we terminate early and return an error (`jv_invalid_with_msg()`)?  Particularly as we're reading the string a byte at a time rather than a Unicode codepoint at a time.